apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/transaction/types/ms/MSReserveIncreaseTransactionType.java

```diff
@@ -70,16 +70,23 @@ public void doStateIndependentValidation(Transaction transaction) throws AplExce
             throw new AplException.NotValidException("Reserve increase amount must be positive: " + attachment.getAmountPerUnitATM());
         }
     }
-
+// TODO: Check if exception will not be more suitable
     @Override
     public boolean applyAttachmentUnconfirmed(Transaction transaction, Account senderAccount) {
         MonetarySystemReserveIncrease attachment = (MonetarySystemReserveIncrease) transaction.getAttachment();
         Currency currency = currencyService.getCurrency(attachment.getCurrencyId());
-        if (senderAccount.getUnconfirmedBalanceATM() >= Math.multiplyExact(currency.getReserveSupply(), attachment.getAmountPerUnitATM())) {
+        try{
+            if (senderAccount.getUnconfirmedBalanceATM() >= Math.multiplyExact(currency.getReserveSupply(), attachment.getAmountPerUnitATM())) {
             getAccountService().addToUnconfirmedBalanceATM(senderAccount, getLedgerEvent(), transaction.getId(), -Math.multiplyExact(currency.getReserveSupply(), attachment.getAmountPerUnitATM()));
             return true;
         }
         return false;
+        }
+        catch (java.lang.ArithmeticException e)
+        {
```

Comment:
Add logging here.
put
`@Slf4j`
on top of class,
put logging into catch block:
`log.error("Error: currency = {}, attachment = {}", currency, attachment);`